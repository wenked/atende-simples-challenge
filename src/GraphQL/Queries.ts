import { gql } from '@apollo/client';

export const VEHICLESNAMES_QUERY = gql`
	query {
		allVehicles {
			vehicles {
				name
				id
			}
		}
	}
`;

export const FILMSTITLES_QUERY = gql`
	query {
		allFilms {
			films {
				title
				id
			}
		}
	}
`;

export const PEOPLENAMES_QUERY = gql`
	query {
		allPeople {
			people {
				id
				name
			}
		}
	}
`;

export const ALLDATA_QUERY = gql`
	query {
		allFilms {
			films {
				episodeID
				title
				openingCrawl
				director
				producers
				releaseDate
				id
			}
		}
		allPeople {
			people {
				id
				name
				birthYear
				eyeColor
				gender
				hairColor
				height
				mass
				skinColor
				homeworld {
					name
				}
				species {
					name
				}
				created
			}
		}
		allVehicles {
			vehicles {
				id
				name
				model
				vehicleClass
				manufacturers
				costInCredits
				length
				crew
				passengers
				maxAtmospheringSpeed
				cargoCapacity
				consumables
			}
		}
	}
`;

export const FILMS_QUERY = gql`
	query {
		allFilms {
			films {
				episodeID
				title
				openingCrawl
				director
				producers
				releaseDate
				created
				edited
				id
			}
		}
	}
`;

export const PEOPLE_QUERY = gql`
	query {
		allPeople {
			people {
				id
				name
				birthYear
				eyeColor
				gender
				hairColor
				height
				mass
				skinColor
				homeworld {
					name
				}
				species {
					name
				}
				created
			}
		}
	}
`;

export const VEHICLES_QUERY = gql`
	query {
		allVehicles {
			vehicles {
				id
				name
				model
				vehicleClass
				manufacturers
				costInCredits
				length
				crew
				passengers
				maxAtmospheringSpeed
				cargoCapacity
				consumables
			}
		}
	}
`;
