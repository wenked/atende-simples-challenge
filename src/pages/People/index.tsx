import React, { useState, useEffect } from 'react';
import { useQuery } from '@apollo/client';
import { PEOPLENAMES_QUERY } from '@graphql/Queries';
import { peopleInterface } from 'src/types';
import { ItemListContainer } from '@components/Layout/ItemListContainer/styles';
import { ListContainer } from '@components/Layout/ListContainer/styles';
import { LoadingContainer } from '@components/Layout/LoadingContainer/styles';

const People: React.FC = () => {
	const { loading, error, data } = useQuery(PEOPLENAMES_QUERY);
	const [people, setPeople] = useState<peopleInterface[]>();

	useEffect(() => {
		if (data) {
			if (data) {
				setPeople(data.allPeople.people);
			}
		}
	}, [data]);

	return error ? (
		<h2>Failed to fetch data</h2>
	) : loading ? (
		<LoadingContainer>
			<h2>Loading...</h2>
		</LoadingContainer>
	) : (
		<ListContainer>
			{people?.map((person) => {
				return (
					<ItemListContainer key={person.id}>{person.name}</ItemListContainer>
				);
			})}
		</ListContainer>
	);
};

export default People;
