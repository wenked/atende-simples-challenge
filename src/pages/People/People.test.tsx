import { MockedProvider } from '@apollo/client/testing';
import { mount, shallow } from 'enzyme';
import React from 'react';

import { PEOPLENAMES_QUERY } from '@graphql/Queries';
import { render } from '@testing-library/react';
import { act } from 'react-dom/test-utils';
import People from './index';

const mocks = [
	{
		request: {
			query: PEOPLENAMES_QUERY,
		},
		result: {
			data: {
				allPeople: {
					people: [
						{
							name: 'Luke Skywalker',
							id: 'cGVvcGxlOjE=',
						},
						{
							name: 'C-3PO',
							id: 'cGVvcGxlOjI=',
						},
						{
							name: 'R2-D2',
							id: 'cGVvcGxlOjM=',
						},
						{
							name: 'Darth Vader',
							id: 'cGVvcGxlOjQ=',
						},
					],
				},
			},
		},
	},
];

describe('rendering component', () => {
	it('rendering without crashing', async () => {
		let wrapper = mount(
			<MockedProvider mocks={mocks} addTypename={false}>
				<People />
			</MockedProvider>
		);
		console.log(wrapper.debug());
		await act(async () => {
			await new Promise((resolve) => setTimeout(resolve, 0));
		});
		wrapper.update();
		console.log(wrapper.debug());

		expect(
			wrapper.contains(<div className='sc-bdfBwQ vBmnt'>Darth Vader</div>)
		).toEqual(true);
	});
});

test('snapshot', async () => {
	const { container } = render(
		<MockedProvider
			mocks={mocks}
			addTypename={false}
			defaultOptions={{ watchQuery: { fetchPolicy: 'no-cache' } }}>
			<People />
		</MockedProvider>
	);

	await act(async () => {
		await new Promise((resolve) => setTimeout(resolve, 1000));
	});

	expect(container.firstChild).toMatchSnapshot();
});
