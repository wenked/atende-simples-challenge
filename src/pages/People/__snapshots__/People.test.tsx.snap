// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div
  class="sc-gsTCUz uRvXt"
>
  <div
    class="sc-bdfBwQ vBmnt"
  >
    Luke Skywalker
  </div>
  <div
    class="sc-bdfBwQ vBmnt"
  >
    C-3PO
  </div>
  <div
    class="sc-bdfBwQ vBmnt"
  >
    R2-D2
  </div>
  <div
    class="sc-bdfBwQ vBmnt"
  >
    Darth Vader
  </div>
</div>
`;
